common-libs/common/src/Affinity.ts

```diff
@@ -624,21 +625,21 @@ export class Affinity {
 
   private _buildFragment<TKeys extends string, TData extends SimpleThing & Record<TKeys, unknown>>(
     credential: VCV1<VCV1Subject<TData>>,
-    fields: TKeys[],
+    fields: string[],
   ) {
     if (Array.isArray(credential.credentialSubject)) {
       throw new Error()
     }
 
-    const dataFields: Record<TKeys, Record<string, never>> = {} as any
-    for (const field of fields) {
-      if (credential.credentialSubject.data[field] === undefined) {
-        throw new Error(`Field "${field}" not a part of credential`)
-      }
-
-      dataFields[field] = {}
+    const validationErrors = validateObjectHasPaths(credential.credentialSubject.data, fields)
+    if (validationErrors) {
+      const firstError = validationErrors[0]
+      throw new Error(`Field "${firstError.field}" not a part of credential`)
```

Comment:
Ideally the error would contain all missing fields, not just the first one that's missing. It's not good developer experience if you get errors one by one.